kernel/tilck_cmd.c

```diff
@@ -63,6 +65,13 @@ static int sys_tilck_run_selftest(const char *u_selftest)
    return se_run(se);
 }
 
+static int sys_busy_wait(ulong n)
```

Comment:
Naming: this is not a Linux syscall. Therefore, its name shouldn't start with `sys_`. Make it be something like `tilck_busy_wait()`.

Btw, this whole function is set-only and it should be placed in modules/systests/systests.c and registered with `register_tilck_cmd()`. See the other "test" syscalls in systests.c.

Reply:
No problem! Though it's not clear to me the benefit in only loading these "tilck command" functions lazily. Is it to avoid loading their code?

Reply:
@cozis Good question. The benefit is to not introduce test-related code in the kernel which not only increases the code size, but can be a security risk. If we imagine what "production Tilck" could be, will it have those test functions? It's much better to keep the non-essential code outside of the *core* kernel. That's why we have modules and the kernel self tests in `tests/self` are also compiled conditionally when KERNEL_SELFTESTS is 1. That way is super simple to build a minimal Tilck. Consider that not only tests, but all the non-essential code, including the console, the ACPI support, the PS/2 keyboard driver and other stuff are source modules (we don't have loadable modules at runtime).

**Important** point about the test functions: while your function is totally fine if ends up in production, what about `TILCK_CMD_CALL_FUNC_0` which allows userspace to call an arbitrary kernel function **without** any checks? The idea is that in the *test* syscalls we can do _ANYTHING_. No security, no checks to avoid crashes, nothing. Testing can do anything and that's correct. We can also do live-patching and any kind of tricks. But in the actual product, those things shouldn't exist.

In the actual kernel, we can only have *real* Tilck syscalls. So far we have none of such Tilck-only syscalls. Anything in `tilck_cmds` is test-related stuff.  Actually, now I'm thinking that ideally all the function pointers in the `tilck_cmds` array should be NULL. That's not the case only for historical reasons.

The modules are relatively new in Tilck. I introduced them when decided to experiment with the "minimal Tilck" and see how small this kernel get and continue to work. So far, the minimal configuration can run in a i686 QEMU VM with just 3 MB of RAM and have 500-600 KB free. That's good, but I can could make it smaller, if I removed 600 KB for `busybox` and maybe 60-70 KB by removing the self tests. Just, it wouldn't be testable, but for true embedded it will great. Once the whole thing starts to run on ARM64, then a porting to ARM32 might be necessary, as the Cortex series is 32-bit only I think. Then, we can super-creative. Consider that for true embedded we might completely skip the bootloader, busybox and a ton of stuff. To some degree, things will be easier and quite simpler, depending on what we wanna support. But I wanted to have a "friendly" OS that can also run "regular" stuff not just some micro-embedded real-time OS, that's why we're here, trying to do both.

Reply:
Got it. I wasn't thinking about the bigger picture but only about my function which is innocuous. 

Do you think that having modules that load at runtime in Tilck makes sense? Would adding them increase complexity too much?